### PR TITLE
add gravitational potential

### PR DESCRIPTION
**Pull request by simonbyrne**
Ideally we would make use of the topology structure here, but I can't think of a good way to do that in the current configuration. This seems like the easiest option for now.

### CODE REVIEW

**Comment by simonbyrne** [comment]
@charleskawczynski and @asraero pointed out that `RectangularSpace` was not the most informative name.

Basically this determines the orientation of the global forces (gravity / Coriolis) so how about `SphericalOrientation`/`FlatOrientation`?

File: src/Atmos/Model/orientation.jl [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+struct SphericalOrientation <: Orientation
+end
+function init_aux!(m::SphericalOrientation, aux::Vars, x)
+  normcoord = norm(aux.coord)
+  aux.orientation.Φ = grav * (normcoord - planet_radius)
+  aux.orientation.∇Φ = grav / normcoord .* aux.coord
+end
+
+"""
+    SphericalOrientation <: Orientation
```
Comment (lcw) [inline src/Atmos/Model/orientation.jl]:
```suggestion
    FlatOrientation <: Orientation
```

**Pull request merged**